shared/src/androidMain/kotlin/dev/johnoreilly/confetti/work/SessionNotificationWorker.kt

```diff
@@ -0,0 +1,42 @@
+package dev.johnoreilly.confetti.work
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequest
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import dev.johnoreilly.confetti.work.SessionNotificationSender.Companion.INTERVAL
+import kotlin.time.toJavaDuration
+
+class SessionNotificationWorker(
+    context: Context,
+    workerParams: WorkerParameters,
+    private val notifier: SessionNotificationSender,
+) : CoroutineWorker(context, workerParams) {
+
+    override suspend fun doWork(): Result {
+        notifier.sendNotification()
+        return Result.success()
+    }
+
+    companion object {
+
+        private val TAG = SessionNotificationWorker::class.java.simpleName
+
+        fun startWorkRequest(workManager: WorkManager) {
+            workManager.enqueueUniquePeriodicWork(
+                TAG,
+                ExistingPeriodicWorkPolicy.UPDATE,
+                createPeriodicWorkRequest(),
+            )
+        }
+
+        private fun createPeriodicWorkRequest(): PeriodicWorkRequest =
```

Comment:
Running this periodically seems wasteful. Instead could we use the bookmark toggle action to create a once off action that will send the notification at that point. Use the daily job to sync them.

Longer term, we may want to consider sending the notifications via Firebase from the server.

Reply:
> Running this periodically seems wasteful.

Agreed.

> Instead could we use the bookmark toggle action to create a once off action that will send the notification at that point.

Do you mean, whenever the user "adds or removes a new bookmark" we trigger the notifications? If yes, I think that may not be that helpful. The use case I have in mind is:
- I add bookmarks for the conference during the week.
- On the event day, while I'm distracted, the app would notify me a given session is to start so I can go to the room.

FYI: The photo above shows all bookmarked sessions because I removed the filter during my tests but in the code, we [check only the upcoming sessions in a range of up to 15 minutes](https://github.com/joreilly/Confetti/pull/544/files#diff-69b1d1f49e0d1bd139663b6ae3805ea0c9c2d2f8b047a08d104bbfefead356a9R65-R68).

In other words, I would receive notifications throughout the day whenever a session is 15 minutes about to start without requiring me to interact with the app as the room is visible.

> Longer term, we may want to consider sending the notifications via Firebase from the server.

Agreed.

---

Let me know what you think!

Reply:
> Do you mean, whenever the user "adds or removes a new bookmark" we trigger the notifications? If yes, I think that may not be that helpful

No, I meant we don't need to run a job every 15 minutes to create notifications for the next 15 minutes.

We can create jobs scheduled to run when the notification should show. Update these jobs users add or remove favourites.

Reply:
Ahhh, I see. That makes a lot of sense, we should definitely do that.